### DISCUSSION TITLE
how does Slidev compare with Marp?

### DISCUSSION THREAD

**Discussion by olets** [slidevjs/slidev#86] [General]
I'd be interested to hear your take on how Slidev compares to [Marp](https://github.com/marp-team/marp)

**Comment by eggplants** [comment]
Comparison based on my personal point of view:

|term|Marp|Slidev|
|---|---|---|
|Since|[Mar 25, 2018 –](https://github.com/marp-team/marp/graphs/contributors)|[Apr 11, 2021 –](https://github.com/slidevjs/slidev/graphs/contributors)|
|Core Version|[v3.2.1](https://github.com/marp-team/marp-core/releases/tag/v3.2.1)|[v0.33.1](https://github.com/slidevjs/slidev/releases/tag/v0.33.1)|
|Org Stars|[7000~](https://github.com/marp-team)|[22000~](https://github.com/slidevjs)|
|Written|almost vanilla TS|Vue3 with TS (on Vite)|
|CSS Framework|PostCSS|WindiCSS|
|Built-in theme|[3](https://github.com/marp-team/marp/blob/2567f8498280a4d81cc062bf3c934171e1d67b33/website/components/top/Features.tsx#L155-L156)|[5](https://sli.dev/themes/gallery.html#official-themes)|
|Auto-install 3rd-party theme|no|[yes](https://sli.dev/themes/use.html)|
|Camera View|no|[yes](https://sli.dev/guide/recording.html#camera-view)|
|Recording|no|[yes](https://sli.dev/guide/recording.html#recording-1)|
|KaTeX|[yes](https://github.com/marp-team/marp/blob/main/website/docs/guide/math-typesetting.md#math-typesetting)|[yes](https://sli.dev/guide/syntax.html#latex)|
|VS Code extension|[yes](https://github.com/marp-team/marp-vscode)|[yes](https://github.com/slidevjs/slidev-vscode)|
|Repo management|[separated](https://marp.app/#get-started)|[monolithic](https://github.com/slidevjs/slidev/tree/main/packages)|
|Mermaid|[yes (unreliable?)](https://github.com/marp-team/marp/discussions/207#discussioncomment-1594903)|[yes](https://github.com/slidevjs/slidev/blob/a62a40e5a72041d2b61abc3cc22b16d606bc2f9d/README.md?plain=1#L89)|
|Animation|[yes](https://marp.app/blog/how-to-make-custom-transition)|[yes](https://sli.dev/guide/animations.html)|
|...|...|...|
|Almost features|same as Slidev|same as Marp|

I could find no reference to marp anywhere in slidev, but slidev's features clearly cover and enhance marp's.

**Comment by codingluke** [comment]
I think the biggest difference is, that marp started as a open source web version of [deckset](https://www.deckset.com/). Therefore it added all the handy image derictives. https://marpit.marp.app/image-syntax. They are very powerful to build complex presentations with almost no html needed. It is very common Markdown centered.

Sli.dev on the other hand focuses heavily on Vue and html for layouting. It is far more powerfull than marp, however there is much more html/vue code needed, which makes the files a little more cluttered.

As an [example here](https://github.com/codingluke/bbzbl-modul-431/tree/main/slides) are some presentations I did with marp for a modul. Every presentation has only one markdown file. To build all presentations there is only one [cli call](https://github.com/codingluke/bbzbl-modul-431/blob/main/package.json#LL12C81-L12C81) needed. The result can be seen [here](https://codingluke.github.io/bbzbl-modul-431/slides/).

On the other hand sli.dev looks more like a **JS project per presentation**. https://github.com/slidevjs/slidev/tree/main/demo/starter.

My takeaway: When you like interactions animations and a lot of 🤩, use slidev. When you want simple clean slides to focus on the content, use marp.

Another thought: sli.dev is far more active, let's see what the future brings.

**Comment by BretFisher** [comment]
I tried Marp for a few months on some significant projects (50-200 slides) and found it to limiting, so I gave up on it and then found Slidev as a better alternative (I've also used Remark for years on large slide decks). I was looking for a remark replacement with a more modern approach, and Slidev has been what I've settled on this year.

I agree with @codingluke here that Slidev results in larger markdown files with all the extras and frontmatter I need to sprinkle in (compared to Remark), but the community here is excellent and I'm slowly figuring out solutions to all the edge cases I need to solve with it.
